### PR TITLE
chore(e2e): add `input.spec.ts`

### PR DESCRIPTION
**Pull request by splincode**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring
- [ ] Build or CI related changes
- [ ] Tests related changes
- [ ] Documentation content changes

### CODE REVIEW

File: projects/demo-playwright/tests/kit/input/input.spec.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
+import {TuiDocumentationPagePO, tuiGoto} from '@demo-playwright/utils';
+import {expect, test} from '@playwright/test';
+
+test.describe('Input', () => {
+    test("has custom content (text) + cleaner + hint which don't overlapping each others", async ({
+        page,
+    }) => {
+        await tuiGoto(
+            page,
+            '/components/input/API?tuiMode=null&tuiTextfieldCustomContent=<span>LongTextContent<%2Fspan>&tuiHintContent=Some%20content',
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/input/input.spec.ts:10]:
```suggestion
            '/components/input/API?tuiTextfieldCustomContent=<span>LongTextContent<%2Fspan>&tuiHintContent=Some%20content',
```

File: projects/demo-playwright/tests/kit/input/input.spec.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        );
+        await expect(new TuiDocumentationPagePO(page).apiPageExample).toHaveScreenshot(
+            '01-input.png',
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/input/input.spec.ts:13]: Why not `01-custom-text-content-cleaner-hint.png` (as it was in cypress tests)?
(the same question for screenshot name)

File: projects/demo-playwright/tests/kit/input/input.spec.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    });
+
+    test('input overflow due to placeholder', async ({page}) => {
+        await tuiGoto(
+            page,
+            'components/input/API?tuiTextfieldIconLeft=tuiIconSearch&pseudoFocus=true&placeholder=Lorem%20ipsum%20dolor%20sit%20amet,%20consectetur%20adipiscing%20elit,%20sed%20do%20eiusmod%20tempor%20incididunt%20ut%20labore',
+        );
+
+        const example = new TuiDocumentationPagePO(page).apiPageExample;
+        const input = example.locator('.t-input-wrapper').locator('input[tuiTextfield]');
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/input/input.spec.ts:60]: Does example contains only one `input[tuiTextfield]` ?

Could we shorten this line in the following way?
```suggestion
        const input = example.locator('input[tuiTextfield]');
```

File: projects/demo-playwright/tests/kit/input/input.spec.ts [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    });
+
+    test('character descenders overflow the line', async ({page}) => {
+        await tuiGoto(
+            page,
+            '/components/input/API?&pseudoFocus=true&placeholder=big, placeholder, qwerty, jackson, yellow and more',
+        );
+
+        const example = new TuiDocumentationPagePO(page).apiPageExample;
+        const input = example.locator('.t-input-wrapper').locator('input[tuiTextfield]');
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/input/input.spec.ts:78]:
```suggestion
        const input = example.locator('input[tuiTextfield]');
```

File: projects/demo-playwright/tests/kit/input/input.spec.ts [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+    test('external mask works', async ({page}) => {
+        await tuiGoto(page, '/components/input');
+
+        const example = new TuiDocumentationPagePO(page).getExample('#mask');
+        const input = example.locator('tui-input input[tuiTextfield]');
+
+        await example.scrollIntoViewIfNeeded();
+        await input.first().fill('111111111111');
+        await input.last().fill('111111111111');
+        await input.last().blur();
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/input/input.spec.ts:115]: `const inputs = ...` ?

File: projects/demo-cypress/cypress/tests/kit/input/input.cy.ts [hunk 6]
```diff
@@ -1,136 +0,0 @@
...
-    });
-
-    it('can be horizontally scrolled', () => {
-        cy.viewport(450, 300).tuiVisit('components/input/API');
-
-        cy.get('#demo-content').should('be.visible').as('wrapper');
-        cy.get('@wrapper').find('tui-input input[tuiTextfield]').as('input');
-
-        cy.get('@input')
-            .should('not.have.css', 'text-overflow', 'ellipsis')
```
Comment (nsbarsukov) [inline projects/demo-cypress/cypress/tests/kit/input/input.cy.ts:101]: I dont see this check in new test

File: projects/demo-playwright/tests/kit/input/input.spec.ts [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+        await input.focus();
+        await page.keyboard.type(
+            "It has been the industry's standard dummy text ever since the 1500s",
+        );
+        await page.keyboard.press('Enter');
+        await input.blur();
+        await expect(example).toHaveScreenshot('08-input.png');
+    });
+
+    test('can be horizontally scrolled', async ({page}) => {
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/input/input.spec.ts:91]: This test does not have any action with scroll (see previous cypress test).
This test does not complete its main goal (according to title).

**Review by nsbarsukov — CHANGES_REQUESTED** [review]

File: projects/demo-playwright/tests/kit/input/input.spec.ts (hunk 7 above)
Comment (splincode) [reply projects/demo-playwright/tests/kit/input/input.spec.ts:93]: fixed

File: projects/demo-cypress/cypress/tests/kit/input/input.cy.ts (hunk 6 above)
Comment (splincode) [reply projects/demo-cypress/cypress/tests/kit/input/input.cy.ts:101]: fixed

File: projects/demo-playwright/tests/kit/input/input.spec.ts (hunk 5 above)
Comment (splincode) [reply projects/demo-playwright/tests/kit/input/input.spec.ts]: fixed

File: projects/demo-playwright/tests/kit/input/input.spec.ts (hunk 4 above)
Comment (splincode) [reply projects/demo-playwright/tests/kit/input/input.spec.ts]: fixed

File: projects/demo-playwright/tests/kit/input/input.spec.ts (hunk 3 above)
Comment (splincode) [reply projects/demo-playwright/tests/kit/input/input.spec.ts]: fixed

File: projects/demo-playwright/tests/kit/input/input.spec.ts (hunk 2 above)
Comment (splincode) [reply projects/demo-playwright/tests/kit/input/input.spec.ts]: fixed

File: projects/demo-playwright/tests/kit/input/input.spec.ts [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+        await expect(example).toHaveScreenshot('09-horizontally-scrolled.png');
+        await input.focus();
+        await expect(example).toHaveScreenshot('10-horizontally-scrolled.png');
+        await page.evaluate(() => {
+            const input = document.querySelector('input[tuiTextfield]');
+
+            if (input) {
+                input.scrollLeft = input.clientWidth / 2;
+            }
+        });
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/input/input.spec.ts:113]: Is it better to use `await mouse.wheel(deltaX, deltaY)` ?

https://playwright.dev/docs/api/class-mouse#mouse-wheel

See this question:
* https://github.com/microsoft/playwright/issues/13189#issuecomment-1083569970

**Review by nsbarsukov — APPROVED** [review]

File: projects/demo-playwright/tests/kit/input/input.spec.ts (hunk 8 above)
Comment (splincode) [reply projects/demo-playwright/tests/kit/input/input.spec.ts:113]: so difficult to calculate deltaX, deltaY

**Pull request merged**